### PR TITLE
Fixes #8454 Changes to LombScargle-periodogram-methods-docs

### LINKED ISSUE #8454: Negative powers in Lomb-Scargle periodogram
I've been using Astropy's Lomb-Scargle periodogram for some time but just run into an unexpected scenario. I've attached a simple series of 19440 points of Gaussian noise ([WN_1000_timeseries.txt](https://github.com/astropy/astropy/files/2901007/WN_1000_timeseries.txt)), generated using `1000*numpy.random.randn(19440)`. When I compute the Lomb-Scargle periodogram, however, the second-last element has negative power. I was under the impression that this wasn't possible but I appreciate that I may well be wrong, in which case this issue can be resolved by me understanding the LS periodogram better!

The timeseries is attached and this snippet of code produces the negative value for me.

```python
import numpy as np
from astropy.stats import LombScargle

y = np.loadtxt('WN_1000_timeseries.txt')
f, p = LombScargle(np.arange(len(y), dtype=float), y, center_data=False).autopower(normalization='psd', samples_per_peak=1, nyquist_factor=1)
print(p[-2]) # -2493.908823582499
```

I'm running Python 3.7.2 on Fedora 29, with NumPy 1.16.0 and Astropy 3.1.2.

### PR DESCRIPTION
**Pull request by disha012**
closes #8454

### CODE REVIEW

**Comment by pllim** [comment]
Doctest failure seems unrelated.

File: docs/stats/lombscargle.rst [hunk 1]
```diff
@@ -679,6 +679,8 @@ the periodogram frequencies using a fast Fourier transform. As with the
 ``slow`` method, it can handle data errors and floating mean.
 The scaling is approximately :math:`O[N\log M]` for :math:`N`
 data points and :math:`M` frequencies.
+Due to the approximation, negative powers are produced for frequencies close to zero.
+Hence, it is suggested to use other methods in such cases.
```
Comment (jakevdp) [inline docs/stats/lombscargle.rst]: I think this should be reworded. Negative powers are not produced every time, and they don't only happen at frequencies near zero.

Maybe something like "The fast algorithm trades accuracy for speed, and produces a close approximation to the true periodogram. In particular, you may observe powers less than zero in some cases."

File: docs/stats/lombscargle.rst [hunk 2]
```diff
@@ -679,6 +679,8 @@ the periodogram frequencies using a fast Fourier transform. As with the
 ``slow`` method, it can handle data errors and floating mean.
 The scaling is approximately :math:`O[N\log M]` for :math:`N`
 data points and :math:`M` frequencies.
+The fast method trades accuracy for speed, and produces a close approximation to 
+the true periodogram. Specifically, negative powers are obserevd in some cases.
```
Comment (pllim) [inline docs/stats/lombscargle.rst]:
```suggestion
the true periodogram. Specifically, negative powers are observed in some cases.
```

**Comment by astrofrog** [comment]
By the way, we should make sure #8591 is merged first

**Comment by bsipocz** [comment]
Given the timeseries/periodogram restructuring in 3.2 I won't backport this.

**Comment by bsipocz** [comment]
Thanks @disha012!

**Pull request merged**